### PR TITLE
Field Period New/Edit Page Validation & Improvement

### PR DESCRIPTION
**Pull request by matthewandrewpalmer**
- Max 4 character Validation
- Page Design Changes
- Autofocus on Field

### CODE REVIEW

File: blaise_survey_manager/survey_config/templates/edit_field_period.html [hunk 1]
```diff
@@ -5,48 +5,46 @@
...
+                                    </label>
+                                    {{ field(autofocus="autofocus", class_=" input input--text input-type__input form-control is-invalid", autocomplete="new-password") }}
                                     {% for error in field.errors %}
                                         <div class="invalid-feedback">{{ error }}</div>
                                     {% endfor %}
                                 {% else %}
-                                    {{ field(class_="form-control", autocomplete="new-password") }}
+                                    <label class="label" for="text">{{ field.label }}
+                                        <br>
+                                        <span class="label__description u-fs-s">Format Should be MMYY</span>
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/templates/edit_field_period.html]: Insetad of a label, could we have an MMYY placeholder in the box?

File: blaise_survey_manager/survey_config/forms.py [hunk 2]
```diff
@@ -57,6 +57,11 @@ def validate(self):
             log.debug('Cannot validate form: ' + json.dumps(self.errors))
             return False
 
+        if len(self.stage.data) > 4:
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/forms.py]: Could you add some validation for the type of characters you can enter? e.g. we only want valid dates - no letter or anything that's more than 12 months.

File: blaise_survey_manager/survey_config/templates/edit_field_period.html [hunk 3]
```diff
@@ -5,48 +5,46 @@
...
-                <div class="float-right">
-                {{ form.save(class_="btn btn--small btn--primary mr-0") }}
-                </div>
-
-            </form>
-        </div>
+                {% endif %}
+            {% endfor %}
+            <br>
+            {{ form.save(class_="btn btn-group__btn") }}
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/templates/edit_field_period.html]: The 'save' button doesn't seem to be aligned with the text box

Comment (matthewandrewpalmer) [reply blaise_survey_manager/survey_config/templates/edit_field_period.html]: I also had that issue when you first load it in, It looks like its that the styling isn't updated immediately. If you do a hard reload it should show up properly.
image

File: blaise_survey_manager/survey_config/forms.py (hunk 2 above)
Comment (matthewandrewpalmer) [reply blaise_survey_manager/survey_config/forms.py]: Added much better validation, that only allows numbers and a valid month

File: blaise_survey_manager/survey_config/templates/edit_field_period.html (hunk 1 above)
Comment (matthewandrewpalmer) [reply blaise_survey_manager/survey_config/templates/edit_field_period.html]: label is part of the style guide and i've added the placeholder

File: blaise_survey_manager/survey_config/templates/edit_field_period.html (hunk 3 above)
Comment (cadmiumcat) [reply blaise_survey_manager/survey_config/templates/edit_field_period.html]: Good point :) thanks - it looks way better!

File: blaise_survey_manager/survey_config/forms.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
+import re
+
 from flask_wtf import FlaskForm
 from wtforms import StringField, SubmitField, BooleanField, validators, SelectField, HiddenField
 from wtforms.validators import Length
-from wtforms.fields.html5 import DateField
+from wtforms.fields.html5 import DateField, IntegerField
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/forms.py]: IntegerField no longer needed

File: blaise_survey_manager/survey_config/views.py [hunk 5]
```diff
@@ -117,6 +117,7 @@ def add_field_period():
         db.session.add(field_period)
         db.session.commit()
         log.info("Field period: " + field_period.stage + " added")
+        flash('Created New Field Period: ' + field_period.stage, "success")
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/views.py]: Can we change this to  "You have created a new field period:"? thank you :)

File: blaise_survey_manager/survey_config/views.py [hunk 6]
```diff
@@ -133,6 +134,7 @@ def edit_field_period():
         form.populate_obj(field_period)
         db.session.add(field_period)
         db.session.commit()
+        flash('Edited Field Period: ' + field_period.stage, "success")
```
Comment (cadmiumcat) [inline blaise_survey_manager/survey_config/views.py]: Similarly... "You have edited a field period: "

**Pull request merged**